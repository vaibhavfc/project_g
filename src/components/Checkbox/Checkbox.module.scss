@import '../../sass/theme/default';

.root {
  color: $checkbox-square;

  &:hover {
    background-color: rgba($checkbox-hover-unchecked, 0.08);
  }

  &:focus {
    background-color: rgba($checkbox-hover-unchecked, 0.08);
    transition: none;
  }
}

.checked {
  color: $checkbox-checked;

  &:hover {
    background-color: rgba($checkbox-hover-checked, 0.08);
  }

  &:focus {
    background-color: rgba($checkbox-hover-checked, 0.08);
    transition: none;
  }
}

.disabledChecked {
  color: $checkbox-disabled;
}

.label {
  transform: translateX(-4px);
  @include body-2;
  color: $checkbox-label;
  font-size: 0.875rem;
  line-height: 1.25rem;
}
