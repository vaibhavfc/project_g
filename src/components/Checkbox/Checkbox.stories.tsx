// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Checkbox from '.';

import { CheckboxProps } from './Checkbox.type';

export default {
  title: 'Components/Checkbox',
  component: Checkbox,
  parameters: {
    controls: {
      // disabled: true,
    },
  },
} as Meta;

const Template: Story<CheckboxProps> = (args) => <Checkbox {...args} />;

export const Default = Template.bind({});
Default.args = {
};

export const DisabledAndChecked = Template.bind({});
DisabledAndChecked.args = {
  checked: true,
  disabled: true,
};

export const DisabledAndUnChecked = Template.bind({});
DisabledAndUnChecked.args = {
  disabled: true,
};

export const DefaultWithLabel = Template.bind({});
DefaultWithLabel.args = {
  label: 'Checkbox Item',
};

export const DisabledAndCheckedWithLabel = Template.bind({});
DisabledAndCheckedWithLabel.args = {
  label: 'Checkbox Item',
  checked: true,
  disabled: true,
};

export const DisabledAndUnCheckedWithLabel = Template.bind({});
DisabledAndUnCheckedWithLabel.args = {
  label: 'Checkbox Item',
  disabled: true,
};
