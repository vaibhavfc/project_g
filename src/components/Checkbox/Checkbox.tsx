import React, { FC, forwardRef, useMemo } from 'react';
import { Checkbox as MuiCheckbox, FormControlLabel, Typography } from '@material-ui/core';
import classnames from 'classnames';
import { CheckboxProps, CheckboxStylingProps } from './Checkbox.type';

import classes from './Checkbox.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const CheckboxWithInnerRef: FC<CheckboxProps> = ({
  innerRef = null,
  classes: overrideClasses = {},
  children = 'Checkbox',
  tabIndex = 0,
  label, checked, disabled,
  ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<CheckboxStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );

  // const [clicked, setClicked] = React.useState(false);

  // const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
  //   setClicked(event.target.checked);
  //   return clicked;
  // };

  return (
    <FormControlLabel
      classes={{
        label: classes.label,
      }}
      value="end"
      control={(
        <MuiCheckbox
          disabled={disabled}
          checked={checked}
          tabIndex={tabIndex}
          classes={{
            root: classes.root,
            checked: classnames({
              [classes.disabledChecked]: (disabled && checked),
              [classes.checked]: (checked && !disabled),
            }),
          }}
        />
      )}
      label={label}
      labelPlacement="end"
    />
  );
};

const Checkbox = forwardRef<HTMLButtonElement, Omit<CheckboxProps, 'innerRef'>>(
  (props, ref) => <CheckboxWithInnerRef innerRef={ref} {...props} />,
);
export default Checkbox;
