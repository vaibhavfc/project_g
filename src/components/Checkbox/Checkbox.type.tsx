import { Ref } from 'react'
import { CheckboxProps as MuiCheckboxProps, CheckboxClassKey } from '@material-ui/core/Checkbox'
export interface CheckboxStylingProps extends Partial<Record<CheckboxClassKey, string>> {
}

export interface CheckboxProps extends MuiCheckboxProps {
  innerRef: Ref<HTMLButtonElement>;
  label?: string;
  list?: {title: string, }
}
