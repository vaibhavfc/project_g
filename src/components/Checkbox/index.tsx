import Checkbox from './Checkbox';
import { CheckboxProps } from './Checkbox.type';

export type { CheckboxProps };

export default Checkbox;
