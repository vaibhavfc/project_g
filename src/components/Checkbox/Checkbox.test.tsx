import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Checkbox from './Checkbox';

describe('test', () => {
  it('Renders Checkbox Compoents', () => {
    render(<Checkbox>test</Checkbox>);
  });

  it('Matches Snapshot of Checkbox', () => {
    const tree = renderer.create(<Checkbox> TEST</Checkbox>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
