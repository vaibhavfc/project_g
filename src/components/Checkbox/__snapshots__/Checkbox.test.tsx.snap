// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Should match Snapshot of Checkbox 1`] = `
<label
  className="MuiFormControlLabel-root"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        className="PrivateSwitchBase-input-8"
        data-indeterminate={false}
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
        value="end"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label label MuiTypography-body1"
  >
    Checkbox Item
  </span>
</label>
`;
